nipype/pipeline/plugins/multiproc.py

```diff
@@ -100,3 +177,126 @@ def _report_crash(self, node, result=None):
 
     def _clear_task(self, taskid):
         del self._taskresult[taskid]
+
+    def _submit_job(self, node, updatehash=False):
+        self._taskid += 1
+        try:
+            if node.inputs.terminal_output == 'stream':
+                node.inputs.terminal_output = 'allatonce'
+        except:
+            pass
+
+        self._taskresult[self._taskid] = \
+            self.pool.apply_async(run_node,
+                                  (node, updatehash),
+                                  callback=release_lock)
+        return self._taskid
+
+    def _send_procs_to_workers(self, updatehash=False, graph=None):
+        """ Sends jobs to workers when system resources are available.
+            Check memory (gb) and cores usage before running jobs.
+        """
+        executing_now = []
+
+        # Check to see if a job is available
+        jobids = np.flatnonzero((self.proc_pending == True) & \
+                                (self.depidx.sum(axis=0) == 0).__array__())
+
+        # Check available system resources by summing all threads and memory used
+        busy_memory_gb = 0
+        busy_processors = 0
+        for jobid in jobids:
+            busy_memory_gb += self.procs[jobid]._interface.estimated_memory_gb
+            busy_processors += self.procs[jobid]._interface.num_threads
+
+        free_memory_gb = self.memory_gb - busy_memory_gb
+        free_processors = self.processors - busy_processors
+
+        # Check all jobs without dependency not run
+        jobids = np.flatnonzero((self.proc_done == False) & \
+                                (self.depidx.sum(axis=0) == 0).__array__())
+
+        # Sort jobs ready to run first by memory and then by number of threads
+        # The most resource consuming jobs run first
+        jobids = sorted(jobids,
+                        key=lambda item: (self.procs[item]._interface.estimated_memory_gb,
+                                          self.procs[item]._interface.num_threads))
+
+        logger.debug('Free memory (GB): %d, Free processors: %d',
+                     free_memory_gb, free_processors)
+
+        # While have enough memory and processors for first job
+        # Submit first job on the list
+        for jobid in jobids:
+            logger.debug('Next Job: %d, memory (GB): %d, threads: %d' \
+                         % (jobid, self.procs[jobid]._interface.estimated_memory_gb,
+                            self.procs[jobid]._interface.num_threads))
+
+            if self.procs[jobid]._interface.estimated_memory_gb <= free_memory_gb and \
+               self.procs[jobid]._interface.num_threads <= free_processors:
+                logger.info('Executing: %s ID: %d' %(self.procs[jobid]._id, jobid))
+                executing_now.append(self.procs[jobid])
+
+                if isinstance(self.procs[jobid], MapNode):
+                    try:
+                        num_subnodes = self.procs[jobid].num_subnodes()
+                    except Exception:
```

Comment:
So I found we took the code from: https://github.com/nipy/nipype/blob/master/nipype/pipeline/plugins/base.py#L354
So whatever exception is being looked for here - we are doing the same.